rest/src/main/kotlin/org/flaxo/rest/PlagiarismConfiguration.kt

```diff
@@ -1,17 +1,44 @@
 package org.flaxo.rest
 
 import org.flaxo.model.DataManager
+import org.flaxo.rest.manager.gitplag.GitplagClient
+import org.flaxo.rest.manager.gitplag.GitplagManager
 import org.flaxo.rest.manager.moss.MossManager
+import org.flaxo.rest.manager.moss.SimpleMossManager
 import org.flaxo.rest.manager.plagiarism.BasicPlagiarismManager
+import org.flaxo.rest.manager.plagiarism.PlagiarismAnalyser
 import org.flaxo.rest.manager.plagiarism.PlagiarismManager
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
 
 @Configuration
 class PlagiarismConfiguration {
 
+    @Bean
+    fun mossManager(dataManager: DataManager,
+                    plagiarismAnalyser: PlagiarismAnalyser
+    ): MossManager = SimpleMossManager(dataManager, plagiarismAnalyser)
+
     @Bean
     fun plagiarismManager(dataManager: DataManager, mossManager: MossManager): PlagiarismManager =
             BasicPlagiarismManager(dataManager, mossManager)
 
+    @Bean
+    fun gitplagManager(gitplagClient: GitplagClient) = GitplagManager(gitplagClient)
+
+    /**
+     * [GitplagClient] bean
+     */
+    @Bean
+    fun gitplagClient(
+            @Value("\${flaxo.gitplag.url}") gitplagUrl: String
+    ): GitplagClient =
+            Retrofit.Builder()
+                    .baseUrl(gitplagUrl)
+                    .addConverterFactory(JacksonConverterFactory.create())
+                    .build()
+                    .create(GitplagClient::class.java)
```

Comment:
Just found a single issue with connection timeouts while calling gitplag `analyze`. By default retrofit uses okhttp client which has 10 seconds read timeout. It means that if some http request takes more than 10 seconds it will fail with `SocketException` being thrown. 

A simple workaround can be applied in order to change the default read timeout. 

```
val okHttpClient = OkHttpClient.Builder() 
    .readTimeout(5, TimeUnit.MINUTES) 
    .build(); 
return Retrofit.Builder() 
    .baseUrl(gitplagUrl) 
    .addConverterFactory(JacksonConverterFactory.create()) 
    .client(okHttpClient) 
    .build() 
    .create(GitplagClient::class.java)
```

It will be even better to extract number of seconds to use as a default timeout for plagiarism analysis to some application property such as `flaxo.gitplag.timeout`.